src/TheBuild/Acquia/GetLatestBackupTask.php

```diff
@@ -90,207 +55,97 @@ class GetLatestBackupTask extends AcquiaTask {
-    $backup_time = new \DateTime();
-
-    foreach ($backups as $backup) {
-      $backup_time->setTimestamp($backup['started']);
-      if ($backup_time > $threshold_time) {
-        $current_backups[] = $backup;
-      }
+    if (!$api_secret = getenv('ACQUIA_CLOUD_API_SECRET')) {
+      $this->log("Couldn't find ACQUIA_CLOUD_API_SECRET env variable.");
+    }
+    if (!$api_key || !$api_secret) {
+      throw new \BuildException("Credentials are required.");
     }
 
-    return $current_backups;
+    return [
+      'key' => $api_key,
+      'secret' => $api_secret,
+    ];
   }
 
   /**
-   * Get the array of backup records from the Acquia Cloud API JSON output.
-   *
-   * Sorts records from oldest to newest.
-   *
-   * @param \PhingFile $file
-   *   Temp file containing the Acquia Cloud API response.
-   *
-   * @return array
-   *   Acquia backup info array.
-   *
-   * @throws \BuildException
-   *
-   * @SuppressWarnings(PHPMD.ShortVariable)
+   * Set Connection to Acquia Cloud using env variables.
    */
-  protected function getBackupRecords(\PhingFile $file) {
-    if ($file->exists()) {
-      $backups = json_decode($file->contents(), TRUE);
-
-      // If the backup records have loaded as an array, and the first record
-      // has the property that we're using, then it is *probably* valid data.
-      if (isset($backups[0]['started'])) {
-
-        // Sort the backups by start time so that the newest is always last.
-        usort($backups, function ($a, $b) {
-          if ($a['started'] == $b['started']) {
-            return 0;
-          }
-          return ($a['started'] < $b['started']) ? -1 : 1;
-        });
-
-        return $backups;
-      }
-      elseif (count($backups) === 0) {
-        // The site might not have been backed up yet.
-        throw new \BuildException('No Acquia Cloud backups found: ' . $file->getCanonicalPath());
-      }
-    }
-    throw new \BuildException('Acquia Cloud backup records could not be loaded from JSON: ' . $file->getCanonicalPath());
+  private function connectAcquiaCloud($credentials) {
+    $connector = new Connector($credentials);
+    return Client::factory($connector);
   }
 
   /**
-   * Download the latest list of backup records from the Acquia Cloud API.
-   *
-   * @param \PhingFile $backups_file
-   *   The file where the downloaded backup should be stored.
-   */
-  protected function downloadBackupRecords(\PhingFile $backups_file) {
-    $json = $this->getApiResponseBody("/sites/{$this->realm}:{$this->site}/envs/{$this->env}/dbs/{$this->database}/backups.json");
-
-    $writer = new \FileWriter($backups_file);
-    $writer->write($json);
+   * Get latest backup from specified environment.
+   */
+  protected function getLatestBackup($client, $environment_uuid) {
+    $backup = new DatabaseBackups($client);
+    $backups = $backup->getAll($environment_uuid, $this->database);
+    $filepath = $this->dir . '/' . $this->env . '_' . $this->database . '_' . 'sql.gz';
```

Comment:
@acidaniel would it make sense to also set the file to load from backups based on this value?

https://github.com/palantirnet/the-build/pull/218/files#diff-339072f52845d316656969cbac2815305c0ccea0ce9de789ba73d78849336067L141-L145

Reply:
@jesconstantine The load from backups task will load any `*sql.gz` file that exists in the `artifact/backups` directory regardless the name.

Reply:
@jesconstantine I would love to see your commit, but I got 404 message :'( (you know why. _Rub salt in the wound_).

Reply:
@jesconstantine I just changed here the extension, I had a typo, please let me know if there something else to do here.

Reply:
@jesconstantine I think I found a way to improve performance by 96% of loading database with only using Linux commands. Here is a bit of background:

1. Using `drupal-load-db` was using a `gunzip -c` command that when I run it it took around **10 minutes** to finish the import
2. Using `<exec executable="ddev"...` took around **56 seconds** to load, but this is not an option since we need to ensure we are outside the container. So this option is discarded.
3. Figured out and reading that by using native Linux commands like `gzip -dc` in combination with a direct `mysql` command takes only **46 seconds**

You can see how I am implementing it really straightforward. So following the general formula about performance improvement in percentage which is:

```
( new - old ) / old * 100%
```

We have:

```
(42 - 600)/600*100= -93
```

What do you think about this?

Reply:
I'm so sorry about posting that link, I wasn't thinking. 😦